test/local/routes/account.js

```diff
@@ -40,30 +40,33 @@ var makeRoutes = function (options, requireMocks) {
   config.signinUnblock = config.signinUnblock || {}
   config.secondaryEmail = config.secondaryEmail || {}
 
-  var log = options.log || mocks.mockLog()
-  var Password = options.Password || require('../../../lib/crypto/password')(log, config)
-  var db = options.db || mocks.mockDB()
-  var customs = options.customs || {
-    check: function () { return P.resolve(true) }
+  const log = options.log || mocks.mockLog()
+  const mailer = options.mailer || {}
+  const Password = options.Password || require('../../../lib/crypto/password')(log, config)
+  const db = options.db || mocks.mockDB()
+  const customs = options.customs || {
+    check: () => { return P.resolve(true) }
   }
-  var checkPassword = options.checkPassword || require('../../../lib/routes/utils/password_check')(log, config, Password, customs, db)
-  var push = options.push || require('../../../lib/push')(log, db, {})
+  const signinUtils = options.signinUtils || require('../../../lib/routes/utils/signin')(log, config, customs, db, mailer)
+  if (options.checkPassword) {
+    signinUtils.checkPassword = options.checkPassword
+  }
+  const push = options.push || require('../../../lib/push')(log, db, {})
```

Comment:
Most of the test changes here are for shuffling around how things are loaded and/or mocked; I was surprised and pleased how little existing test code had to be changed as part of this refactor.